src/display/editor/editor.js

```diff
@@ -92,6 +93,15 @@ class AnnotationEditor {
     this.div.style.zIndex = this.#zIndex;
   }
 
+  setParent(parent) {
+    if (parent === null) {
+      this._serialized = this.serialize();
```

Comment:
It might be overall safer to just do this unconditionally here, however it might not always be necessary?
Unless I'm misunderstanding things, looking at https://github.com/mozilla/pdf.js/blob/67e1c37e0fea06951ece8c457c2b39d2ee2a683f/src/display/editor/annotation_editor_layer.js#L285-L293 it seems that we'd really only need to "cache" the serializable data if the current editor actually exists in the annotationStorage?